### ISSUE TITLE
Critical error in prep data

### ISSUE THREAD

**Issue by bjc45276** [SchmollerLab/Cell_ACDC#420]
### Description:

Hello, I encountered a critical error in prep data. I also get an error after launching segmentation module. I attached the log file for the first issue. I also am attaching screenshots showing my process
[2023-10-10_13-01-50_dataPrep_b6984124-db13-4b65-88b7-78891456440b_stdout.log](https://github.com/SchmollerLab/Cell_ACDC/files/12859926/2023-10-10_13-01-50_dataPrep_b6984124-db13-4b65-88b7-78891456440b_stdout.log)

![image](https://github.com/SchmollerLab/Cell_ACDC/assets/49136068/d1da2ca8-f246-4697-936a-961b4ff10ec2)
![image](https://github.com/SchmollerLab/Cell_ACDC/assets/49136068/d702267f-7d45-4e17-adb5-4ea417dab019)
![image](https://github.com/SchmollerLab/Cell_ACDC/assets/49136068/c06377be-29ba-4e8e-a16a-b93a823c4a23)
![image](https://github.com/SchmollerLab/Cell_ACDC/assets/49136068/0e62014e-4a87-43a7-89d0-174faca962ea)
![image](https://github.com/SchmollerLab/Cell_ACDC/assets/49136068/38e83c76-d0ae-49f0-80b6-231b4c8680ec)

**Comment by ElpadoCan** [comment]
Hi @bjc45276, 

What file extension is your raw microscopy file? What data does it contain? Multiple channels? 

The problem here is that the data that you created has shape `(2208, 2752, 3)` which for Cell-ACDC it must be either 3D z-stacks of 2208 z-slices with YX shape (2752, 3) or 2208 frames (timelapse) with YX shape (2752, 3). Since neither of these options seems to make sense there is something wrong with the conversion to TIF from the raw microscopy file. This is the first issue that we need to fix.

What is the 3 in the data shape? The number of channels? If yes, you need to change the `Order of dimensions` parameter to the value that would correctly extract the data. If you click on the info button besides the parameter you should get some instructions on how to select the correct one. 

Do you have a chance to share the raw microscopy file with me? You can share it with my Google Drive email padovaf@tcd.ie or use the cloud of your choice. 

Once we fix this issue I think that you will have no problems with data prep. 

One last thing, the error in the screenshot of the terminal is different from the error you have in the log file. It seems like you tried to use YeastMate? Unfortunately, YeastMate does not work on Windows because it uses a library written by META that works only on macOS (and they have zero intentions to expand the compatibility to Windows users). If you really need YeastMate we can try to install it but there is no guarantee it will work unfortunately. 

Let me know and thanks for reporting this!

Cheers,
Francesco

**Comment by bjc45276** [comment]
Hello, thank you for your thorough response. I attached the files I used for the data prep, and the original microscopy file (.czi). I used the ZEN software to convert the .czi file to two .jpg files (two channels). I also tried .tif files but got the same error. I only tried this microscopy picture which contains two channels, no timelapse and no z-stack. This is just a test to see if I can get the Cell-ACDC to work with my image files. I am not sure I need to use YeastMate. I have yeast cells stained with PI and eventually I want to do a cell cycle analysis and measure cell size also.

**Comment by ElpadoCan** [comment]
Nice, it sounds like you want to do exactly what Cell-ACDC was designed for! We will get you started don't worry.

Anyway, I think I understand your problem. When you convert to .jpg you are converting grayscale to RGB so the 3 in your data is the RGB dimension. In Cell-ACDC you need to keep the files as grayscale, which means that the only valid option is TIF files. 

Since you have CZI files then you don't need to convert them to TIFF with the ZEN software because module 0 of Cell-ACDC was designed to perform exactly that. It takes raw microscopy files and it converts the channels to tif files and it puts them into the right structure.

Long story short, I will test with the files you sent me and I will report back to you soon. Thanks! 

Cheers,
Francesco

**Comment by ElpadoCan** [comment]
Ok so as I said you cannot start from JPEG files. However, I just tried to start from the CZI file you sent and it works. You need to follow these steps:

1. Place the CZI file in an empty folder without any other file
2. When asked, select `Multiple files, one for each Position`. Note that you need to select `one for each Position` and NOT `one for each channel`. This is because the CZI file contains the channels and Cell-ACDC will separate them automatically, you don't need to separate them before.
3. For the order of dimensions choose `tzc` (see screenshot below) which should be the default.
4. Run the conversion

Note that it doesn't matter how data is structured inside the CZI file as long as you select the right option. With CZI files you will need `Multiple files, one for each Position` if for example you acquire snapshots and you click on Start experiment on the ZEN software for each field of view. If instead, you run timelapse with tiles, then the Zeiss software will save a single CZI file with all the positions inside, in which case you will select `Single file with one or more positions`. These are the only two options you will ever need. Having files separated by channels is a very specific case of some particular microscope, but luckily, this is not the case with Zeiss.

At the end of the creation process, you should get the files structured as you see in the second screenshot attached. This is the typical data structure required by Cell-ACDC. A "Position" is essentially a folder like `Position_1` containing the folder called `Images`. Inside the `Images` folder you will have one TIF file per channel. Every TIF file can contain 2D data, 3D data (either z-stack or 2D over time), or 4D data (3D z-stacks over time) of the same channel. Have a look at the section of our user manual (link below) `Manually create data structure from microscopy file(s)` because it explains how data should be structured (even though you don't need to do it manually).

User Manual: https://github.com/SchmollerLab/Cell_ACDC/blob/main/UserManual/Cell-ACDC_User_Manual.pdf

Let me know how it goes, thanks! 

Cheers,
Francesco

![image](https://github.com/SchmollerLab/Cell_ACDC/assets/55240925/0811ca67-3639-4e0c-8345-edcfdf2db92b)
![image](https://github.com/SchmollerLab/Cell_ACDC/assets/55240925/a8f99b9e-0f5b-4754-adbf-5def151bc5dd)

**Comment by bjc45276** [comment]
Thank you for your help,
[cid:8b6604e8-87c0-42d1-84a8-fc89d53262e9]

I have the segmentation mask showing up on the cells accurately I think, using yeaZ v2,
I have the cells stained with PI to measure fluorescence intensity of nuclei for each cell for cell cycle analysis. Can I do this in cell-ACDC? What I would hopefully want to do in the future is in more high throughput where I input all of my microscope images (maybe 50-100) and get back information on cell size area/size and fluorescent intensity for each cell.


Sincerely,
Benjamin Chadwick
Plant Biology Graduate Student
University of Georgia

Lin Lab:
http://mibo.franklin.uga.edu/directory/people/xiaorong-lin
810-23-2567-984-015


________________________________

**Comment by ElpadoCan** [comment]
Hi Benjamin,

You can definitely do what you mentioned. There are a couple of ways to achieve this depending on whether you plan to visualize the result of the segmentation and correct errors manually or you accept errors and go fully automatic. 

Since you have the cells segmented with YeaZ_v2 I suppose that you successfully segmented the cells. Did you do it with module 2 or 3? The difference is that in module 2 you can segment multiple positions at once, while in module 3 (the GUI where you can correct errors) you do it Position by Position. 

In any case, once you have the segmentation masks, you can compute measurements. There are two ways to do this. One is in module 3 where you can load multiple Positions at once and then when you save you will be asked if you need to save the measurements. At that dialogue, you can set the measurements. You can also set the measurements you want to save from the menu on the top menubar of the GUI `Measurements --> Set measurements`. You will get a dialogue where you can select which measurements you want to save and for which channel, but you also get an info button for each one of the measurements with an explanation of that specific measurement. 

The other way to save measurements is from the menu `Utilities --> Measurements --> Compute measurements for one or more experiments...` on the top menubar of the little launcher that opens when you run acdc. You typically use this second way when you want to re-compute measurements without loading the data into the GUI of module 3. 

Long story short, if you want to go fully automatic you first segment the cells with module 2 and then you can compute measurements with the utility. This way you never load data into the GUI and you trust the output of the segmentation. 

We recently wrote a tutorial that should partially cover the things that I just said. Feel free to give us feedback about the tutorial, the more people tell us what they think the better it will get. Here is the link https://cell-acdc.readthedocs.io/en/latest/getting-started.html

Let me know how it goes, thanks!

Cheers,
Francesco
